### ISSUE TITLE
Undeprecate no-dot syntax for quantile() and nquantile()

### ISSUE THREAD

**Issue by ron-wolf** [JuliaStats/Distributions.jl#1150]
This issue picks up where JuliaStats/StatsBase.jl#586 left off. At issue is the following snippet in the Distributions package:

https://github.com/JuliaStats/Distributions.jl/blob/20c91d9efcc5f96913bf8e38be2e3fb14b21942b/src/deprecates.jl#L39

The macro affects the `quantile()` and `nquantile()` functions, of which the former is used by StatsBase in [this line](https://github.com/JuliaStats/StatsBase.jl/blob/54cc6fe2e2f709623b9c75275884f9b295fdbd35/src/scalarstats.jl#L340).

According to @nalimilan, StatsBase relies on the deprecated behavior. This is because when `quantile()` is broadcasted over the given weights/probabilities (second param, `p::Union{AbstractArray, Tuple{Vararg{Real}}}`), `_quantilesort!` is run twice on the same parameter (first param, `v::AbstractVector`). See [this code in the Statistics package](https://github.com/JuliaLang/Statistics.jl/blob/b384104d35ff0e7cf311485607b177223ed72b9a/src/Statistics.jl#L858-L1053) where the sorting function would be broadcast over using the new syntax requested by Distributions. In any case, requiring broadcasting would mean requiring the first parameter to be sorted twice.

Would it be best to un-deprecate this functionality, or is there another course? For now, using Statistics as per its own documentation gives the user a deprecation warning, which is… not ideal, especially for first-time users, of which I’m one.

**Comment by nalimilan** [comment]
To complete a bit: `quantile(x, p::AbstractArray)` isn't really needed when `x` is a `Distribution` (in Distributions.jl), since it's strictly equivalent to broadcasting. But when `x` is a vector (in StatsBase), broadcasting involves sorting twice as @ron-wolf said. So it would make sense to undeprecate this method in Distributions for consistency.

Another solution would be to define a custom broadcasting method for `quantile` in StatsBase which would only sort once. Not sure whether that's a good idea or not.

**Comment by ron-wolf** [comment]
@nalimilan Sorry for the delay. I think the best solution is just to undeprecate non-broadcast calls to `quantile`, since they seem to be needed. (If I am wrong and they should never be used outside of StatsBase, please let me know, as that will probably require a clever solution. In any case…) I would like to submit a pull request, but how would I actually go about undeprecating this functionality?

**Comment by nalimilan** [comment]
No problem. You just have to remove the `@deprecate` definition in Distributions and replace it with a normal function definition (and move it to a better place).

**Comment by ron-wolf** [comment]
But won’t that undeprecate _all_ of these functions? Is that what we want?

https://github.com/JuliaStats/Distributions.jl/blob/20c91d9efcc5f96913bf8e38be2e3fb14b21942b/src/deprecates.jl#L27-L31

Also, why would we add the definition of `quantile` here in Distributions when it’s already defined in StatsBase? Maybe I’m a bit turned around about what’s defined or deprecated where.

**Comment by nalimilan** [comment]
Right, you should only remove `quantile` from that list (and define it elsewhere). Also, `quantile` isn't defined for `Distribution` objects in StatsBase nor in Statistics AFAIK. Actually, the definition in Statistics will call `collect(d::Distribution)`, which doesn't make sense.

**Comment by ron-wolf** [comment]
Okay. Do you think it’s only `quantile` that should be removed?

**Comment by nalimilan** [comment]
Yes.
